### PR TITLE
Feature/10 reverse lookup

### PR DESCRIPTION
**Pull request by cryptomental**
I had to connect many dots to implement this, but I am happy with the outcome:

image

I will likely need to implement a few tests as this adds a lot of code.

### CODE REVIEW

**Comment by cryptomental** [comment]
I added find action test, reducer test and component test, @eswarasai - pr ready for the review.

**Comment by eswarasai** [comment]
@cryptomental -- The changes look good to me but I haven't tested the functionality yet. I'll update you on this PR today EOD. Sorry about the delay.

**Comment by cryptomental** [comment]
@eswarasai thanks!

**Comment by pelsasser** [comment]
@eswarasai @cryptomental thanks guys - I am travelling so a little out of touch, appreciate you guys keeping the momentum going!

File: src/components/FindContract/FindContractForm.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+          <Col {...formButtonLayout}>
+            <Button className="submit-button" type="primary" htmlType="submit" loading={this.props.loading} disabled={this.isSubmitDisabled()} style={{width: '100%'}}>
+              Find Contract
+            </Button>
+          </Col>
+        </Row>
+
+        <Row type="flex" justify="center" style={{ marginTop: '16px' }}>
+          <Col {...formButtonLayout}>
+            <Button className="reset-button" type="secondary" style={{width: '100%'}} disabled={this.props.loading} onClick={this.handleReset.bind(this)}>
```
Comment (eswarasai) [inline src/components/FindContract/FindContractForm.js]: @cryptomental -- `Reset Form` is not required for this. Can you please remove this?

**Comment by eswarasai** [comment]
@cryptomental -- Overall the changes are lgtm. There are few minor issues with formatting and also the alphabetical ordering of `imports`, `keys` etc. If you can fix these too, it'd be great.

@pelsasser -- Find Contract functionality is in place but displaying the values of the Contract needs some Design focus. Would you like me to address this issue right away or pick it up later?

**Comment by cryptomental** [comment]
Thanks for the review @eswarasai ! I will fix the formatting today evening (Western European time).
And yes, please update the design if needed. I used a grid , but perhaps you will find a nicer way to display data.

**Comment by cryptomental** [comment]
I googled a bit and did a fully automated sort of the imports using https://www.npmjs.com/package/js-import-sort . I will now have a look at the coverage.

**Comment by cryptomental** [comment]
@eswarasai "loss" of the coverage if mostly caused by this:

https://coveralls.io/builds/15859172/source?filename=src%2Factions%2Ffind.js#L13

This is a missing test for actions/find.js , which is also missing for action/explorer.js  , see https://coveralls.io/builds/15859172/source?filename=src/actions/explorer.js

and also for https://coveralls.io/builds/15859172/source?filename=src/components/FindContract/FindContractField.js

I will try to add test(s) for the latter one to cover at least partially the component code..

**Comment by eswarasai** [comment]
@cryptomental -- W.r.t. to coverage, I'd suggest taking feedback from @pelsasser and @perfectmak before proceeding further. I'm currently unaware of the test suite functionalities 😅

**Comment by cryptomental** [comment]
@eswarasai all right! :-) so the imports are sorted now. I hope we can merge.

The tests if needed we can also add later.

**Comment by eswarasai** [comment]
@pelsasser -- Can you give go ahead for merging this?

Pending tasks:
1. Modify the UI to display information of the Contract for `Find Contracts`
2. Add test cases to improve the `coveralls` percentage

**Comment by cryptomental** [comment]
@perfectmak @pelsasser I am waiting then for green light to merge this one ;)

And I finally found a workaround for both MetaMask issues:

If web3 provider is not recognized in Dapp, switch to Main network, log out, log in, switch to localhost:9545
For 'nonce' issue, click in MetaMask on 'settings', choose 'Reset account', and repeat steps above.

**Comment by pelsasser** [comment]
@eswarasai - you want to create those issues and we can get some ETH attached?
@cryptomental - thanks so much will send some ETH - appreciate the hard work on this.

**Pull request merged**